libpod/container_internal.go

```diff
@@ -964,18 +973,18 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 			return nil, errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to add additional groups", c.ID())
 		}
 		for _, group := range c.config.Groups {
-			_, gid, err := chrootuser.GetUser(c.state.Mountpoint, strconv.Itoa(int(g.Spec().Process.User.UID))+":"+group)
+			gid, err := chrootuser.GetGroup(c.state.Mountpoint, group)
 			if err != nil {
 				return nil, err
 			}
-			g.AddProcessAdditionalGid(uint32(gid))
+			g.AddProcessAdditionalGid(gid)
 		}
 	}
 
 	// Look up and add groups the user belongs to
 	groups, err := chrootuser.GetAdditionalGroupsForUser(c.state.Mountpoint, uint64(g.Spec().Process.User.UID))
 	if err != nil {
-		return nil, err
+		logrus.Debugf("user %q doesn't belong to any additional groups: %v", c.config.User, err)
```

Comment:
Can we safely ignore this error always? It seems like this could indicate real problems.

Reply:
I believe so as this is just looking up if the user belongs to any other additional groups and it is possible for the uid to not belong to any other groups. If we return the error instead, it will fail with "unknown user" when the uid is an int provided by the caller.

Reply:
That's a really bad error. Can we improve the chrootuser package to get us a better one?

Reply:
Okay, so if looking up the username associated with the uid fails (because it does not exist), I am logging the error instead and exiting out of the function. If an error happens after that in the actual look up of additional groups, the error is returned. Is that better?

Reply:
Yeah, that sounds reasonable. Or you could return an error we can easily check for, something like chrootuser.ErrNoSuchUser, and we'll discard just that error.

Reply:
Fixed